src/test/clojure/xtdb/operator/scan_test.clj

```diff
@@ -373,66 +375,127 @@
+            end-time (time/instant->micros (time/->instant #inst "2021"))]
+        (t/is (= {:valid-time [start-time end-time],
+                  :system-time [start-time end-time]}
+                 (tu/temporal-bounds->data (->temporal-bounds vw/empty-args
+                                                              '{:for-valid-time [:in
+                                                                                 [:expr (- (current-timestamp) #xt/interval "P12M")]
+                                                                                 [:expr (+ (current-timestamp) #xt/interval "P12M")]]
+                                                                :for-system-time [:in
+                                                                                  [:expr (- (current-timestamp) #xt/interval "P12M")]
+                                                                                  [:expr (+ (current-timestamp) #xt/interval "P12M")]]}
+                                                              snapshot-time)))))
+
+      (let [vt-start-time (time/instant->micros (time/->instant #inst "2019"))
+            vt-end-time (time/instant->micros (time/->instant #inst "2021"))
+            st-start-time (time/instant->micros (time/->instant #inst "2099"))
+            st-end-time (time/instant->micros snapshot-time)]
+        (t/is (= {:valid-time [vt-start-time (inc vt-end-time)],
+                  :system-time [st-start-time (inc st-end-time)]}
+                 (with-open [args (tu/open-args [#inst "2100"])]
+                   (tu/temporal-bounds->data (->temporal-bounds args
+                                                                '{:for-valid-time [:between
+                                                                                   [:expr (- (current-timestamp) #xt/interval "P12M")]
+                                                                                   [:expr (+ (current-timestamp) #xt/interval "P12M")]]
+                                                                  :for-system-time [:between
+                                                                                    [:expr (- ?_0 #xt/interval "P12M")]
+                                                                                    [:expr (+ ?_0 #xt/interval "P12M")]]}
+                                                                snapshot-time))))))
+
+      (t/testing "nil behaviour"
+        (t/is (= [(time/instant->micros clock-time) (time/instant->micros time/end-of-time)]
+                 (-> (->temporal-bounds vw/empty-args
+                                        '{:for-valid-time [:in [:expr nil] [:expr nil]]}
+                                        snapshot-time)
+                     tu/temporal-bounds->data
+                     :valid-time)))))))
+
+;; TODO adapt for scan/->temporal-bounds
+#_(t/deftest can-create-temporal-min-max-range
```

Comment:
I left this test as a TODO as it mainly concerns predicates on temporal columns which get translated into valid-time and system-time constraints if possible.

Reply:
test removed on main :pray: